tests/trainer/connectors/test_signal_connector.py

```diff
@@ -57,3 +59,30 @@ def training_step(self, batch, batch_idx):
         else:
             trainer.fit(model)
         assert trainer._terminate_gracefully == (False if register_handler else terminate_gracefully)
+
+
+@pytest.mark.parametrize("auto_requeue", (True, False))
+def test_auto_requeue_flag(auto_requeue):
+    sigterm_handler_default = signal.getsignal(signal.SIGTERM)
+    sigusr1_handler_default = signal.getsignal(signal.SIGUSR1)
+
+    trainer = Trainer(plugins=[SLURMEnvironment(auto_requeue=auto_requeue)])
+    connector = SignalConnector(trainer)
+    connector.register_signal_handlers()
+
+    if auto_requeue:
+        sigterm_handlers = signal.getsignal(signal.SIGTERM).signal_handlers
+        assert len(sigterm_handlers) == 1
+        assert sigterm_handlers[0].__qualname__ == "SignalConnector.sigterm_handler_fn"
+
+        sigusr1_handlers = signal.getsignal(signal.SIGUSR1).signal_handlers
+        assert len(sigusr1_handlers) == 1
+        assert sigusr1_handlers[0].__qualname__ == "SignalConnector.slurm_sigusr1_handler_fn"
+    else:
+        assert signal.getsignal(signal.SIGTERM) is sigterm_handler_default
+        assert signal.getsignal(signal.SIGUSR1) is sigusr1_handler_default
+
+    # restore the signal handlers so the next test can run with system defaults
+    # TODO: should this be done in SignalConnector teardown?
```

Comment:
Good catch! There should be teardown on the SignalConnector.

Reply:
yep, working on a follow up